src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -669,11 +670,11 @@ private static bool IsWebSocketsSupported()
 
         private async Task<NegotiationResponse> GetNegotiationResponseAsync(Uri uri, CancellationToken cancellationToken)
         {
-            var negotiationResponse = await NegotiateAsync(uri, _httpClient, _logger, cancellationToken);
+            var negotiationResponse = await NegotiateAsync(uri, _httpClient!, _logger, cancellationToken);
             // If the negotiationVersion is greater than zero then we know that the negotiation response contains a
             // connectionToken that will be required to conenct. Otherwise we just set the connectionId and the
             // connectionToken on the client to the same value.
-            _connectionId = negotiationResponse.ConnectionId;
+            _connectionId = negotiationResponse.ConnectionId!;
```

Comment:
Would it be better to throw if there's no ConnectionId in the negotiate response?

Reply:
It already does in `NegotiateProtocol.ParseResponse`
https://github.com/dotnet/aspnetcore/blob/65bcc540fb7ad86124128b123ea4aa52e8c6dd9c/src/SignalR/common/Http.Connections.Common/src/NegotiateProtocol.cs#L227